easybuild/easyconfigs/b/binutils/binutils-2.31.1_fix_gcc10.patch

```diff
@@ -0,0 +1,26 @@
+Widen integer type compared to enum
+
+GCC 10 seems to enforce signedness between types when comparing values.
+
+Author: Lars Viklund <zao@zao.se>
+diff -ru binutils-2.31.1.orig/gold/x86_64.cc binutils-2.31.1/gold/x86_64.cc
+--- binutils-2.31.1.orig/gold/x86_64.cc	2018-06-24 20:38:57.000000000 +0200
++++ binutils-2.31.1/gold/x86_64.cc	2020-05-13 00:56:30.254725766 +0200
+@@ -1585,7 +1585,7 @@
+ {
+   uint32_t val = 0;
+ 
+-  switch (pr_type)
++  switch ((size_t)(uint32_t)pr_type)
```

Comment:
Why the double conversion? Either one is correct and sufficient.

Reply:
Oh and why is it required at all for 2.31.1? `unsigned int pr_type` is already unsigned.

Reply:
Ok, sorry for the noise, the git interface is confusing there. I can confirm that this issue only is in 2.31.x and can be fixed with `switch ((unsigned int)pr_type)`. It is already fixed in 2.32 (checked the archive)

Reply:
Is `switch ((unsigned int)pr_type)` better for any particular reason?

Reply:
Just easier to read and less confusion by the duplicated conversion. Nothing else.
Alternatively one can you the official patch: https://sourceware.org/git/?p=binutils-gdb.git;a=patch;h=dc1f2887c513c4c2110d2d997ab33ce36b9696fc

Reply:
Applying the official patch makes sense... @zao?

Reply:
Yeah, sure. The double conversion is because I first went directly to `(size_t)` but then found that I had to change the sign first before extending.